app/src/main/java/com/example/sharewhatyoucanproject/LocationTrack.kt

```diff
@@ -0,0 +1,250 @@
+import android.provider.Settings
+import com.example.sharewhatyoucanproject.DashboardActivity
+import java.lang.Exception
+
+class LocationTrack(private val mContext: Context) : Service(), LocationListener {
+    var checkGPS = false
+    var checkNetwork = false
+    var canGetLocation = false
+    var loc: Location? = null
+    private var latitude = 0.0
+    private var longitude = 0.0
+    protected var locationManager: LocationManager? = null// TODO: Consider calling
+    //    ActivityCompat#requestPermissions
+    // here to request the missing permissions, and then overriding
+    //   public void onRequestPermissionsResult(int requestCode, String[] permissions,
+    //                                          int[] grantResults)
+    // to handle the case where the user grants the permission. See the documentation
+    // for ActivityCompat#requestPermissions for more details.
+
+
+    /*if (checkNetwork) {
+
+
+           if (ActivityCompat.checkSelfPermission(mContext, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(mContext, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+               // TODO: Consider calling
+               //    ActivityCompat#requestPermissions
+               // here to request the missing permissions, and then overriding
+               //   public void onRequestPermissionsResult(int requestCode, String[] permissions,
+               //                                          int[] grantResults)
+               // to handle the case where the user grants the permission. See the documentation
+               // for ActivityCompat#requestPermissions for more details.
+           }
+           locationManager.requestLocationUpdates(
+                   LocationManager.NETWORK_PROVIDER,
+                   MIN_TIME_BW_UPDATES,
+                   MIN_DISTANCE_CHANGE_FOR_UPDATES, this);
+
+           if (locationManager != null) {
+               loc = locationManager
+                       .getLastKnownLocation(LocationManager.NETWORK_PROVIDER);
+
+           }
+
+           if (loc != null) {
+               latitude = loc.getLatitude();
+               longitude = loc.getLongitude();
+           }
+       }*/
+// if GPS Enabled get lat/long using GPS Services
+    // get GPS status
+
+    // get network provider status
+    private val location: Location?
+        private get() {
+            try {
+                locationManager = mContext
+                    .getSystemService(LOCATION_SERVICE) as LocationManager
+
+                // get GPS status
+                checkGPS = locationManager!!
+                    .isProviderEnabled(LocationManager.GPS_PROVIDER)
+
+                // get network provider status
+                checkNetwork = locationManager!!
+                    .isProviderEnabled(LocationManager.NETWORK_PROVIDER)
+                if (!checkGPS && !checkNetwork) {
```

Comment:
Our goal is to make this _call site_ clearer.  The (re)naming suggestions above should help with that.  For example `if userHasEnabledLocationTracking` is much clearer than `if checkGPS`, right?  After all, you don't mean that you have checked the GPS, but that the answer from that check is `true`, right?